examples/08_deps_with_different_platforms/README.md

```diff
@@ -0,0 +1,37 @@
+# Example 08: Building deps on different platforms
+This example demonstrates how to build dependencies of a rule for a 
+different platform from the rule itself. 
+
+## Commands
+```
+bazel build //examples/08_deps_with_different_platforms:my_rule
+> Running implementation fxn of dep
+> platform from config: //:linux_platform
+> Running implementation fxn of my_rule
+> platform from config: @local_config_platform//:host
+
+bazel build //examples/08_deps_with_different_platforms:my_rule --platforms=//:windows_platform
+> Running implementation fxn of dep
+> platform from config: //:linux_platform
+> Running implementation fxn of my_rule
+> platform from config: //:windows_platform
+
+```
+
+## Description
+Sometimes you want to write a rule that can build its
+dependencies for different platforms. Instead of two 
```

Comment:
There is this friction that platform targets that rule owners provide are not precise enough, and users want to specify platform targets themselves. Wdyt about mentioning that platform targets don't have to be hardcoded, that they can be specified using the build_setting? Or is the result complicated enough that it warrants its own example?

Reply:
Do you mean like users can create their own platforms using a collection of build settings and transition on those build settings? Or do you mean they can set --platforms to another target they define themselves? If the latter, how do the build settings work in? Either way I think this would be a great point to clarify!

Reply:
I think I thought the second :) Say I am an author of the rule that does a transition. I provide some default platform targets that I think make sense and my transition changes --platforms to my default platforms. But users sometimes want to specify their own platforms to transition to.

Let me describe the real world use case. Apple multicpu transition or android fat apk transition have default cpus they transition to, but also allow the user to specify cpus manually (using e.g. --fat_apk_cpus or smth like that). The typical usecase with cpus is that users don't need to build for all the possible android cpus when they are only testing on the simulator. Platforms will replace cpus in these transitions, and users will need a way to specify --fat_apk_platforms. And this is what I had in mind for the example. Does it make sense?

Reply:
Ah, so is what you're suggesting some kind of build setting that is an override for --platforms that users can set themselves? I'm trying to make sense of the relationship between fat_apk_cpu and cpu right now to pull together a relevant example. 

What's to stop users from directly setting --platform (I guess, same question for them directly setting --cpu?) I guess I'm still confused about the value add here, thanks for taking the time to explain!

Reply:
What I'm suggesting is to have a simple example that corresponds to the --fat_apk_cpu use case. --fat_apk_cpu will be replaced by --fat_apk_platforms in the near future, and its transitions will change the value of --platforms instead of the value of --cpu.

In this example:

1) We'll add a new build_setting (corresponding to --fat_apk_cpu) that will be a list of labels, and the default value will be `//:linux_platform,//:windows_platform`. Let's name it `//:yolo_fat_platforms`.
2) We will implement a yolo_fat_binary rule that will build its dependencies for all platforms in `//:yolo_fat_platforms`. So it will create configuration transition for each platform in `//:yolo_fat_platforms`, changing `--platforms` value to the platform from the build setting.
3) Now user can run the same build, and pass `//:linux_platform,//:windows_platform,//:mac_platform` as `//:yolo_fat_platforms`, and Bazel will build dependencies of `yolo_fat_binary` 3 times.

FTR - this whole PR LGTM, feel free to merge and create a separate one for this complicated example.

Reply:
Ahhh, thank you so much for this detailed comment. Super super helpful.